api/src/main/java/org/infinispan/api/common/CacheWriteOptions.java

```diff
@@ -13,18 +12,32 @@ static Builder writeOptions() {
       return new Builder();
    }
 
+   static Builder writeOptions(CacheOptions options) {
+      Builder builder = new Builder();
+      options.timeout().ifPresent(builder::timeout);
+      options.flags().ifPresent(builder::flags);
+      return builder;
+   }
+
+   static Builder writeOptions(CacheWriteOptions options) {
+      Builder builder = writeOptions((CacheOptions) options);
+      options.expiration().lifespan().ifPresent(builder::lifespan);
```

Comment:
Don't we want to invoke the rawTimeout method here to avoid the lambda? Same with the rest of the methods.